rust/worker/src/execution/operators/flush_segment_writer.rs

```diff
@@ -0,0 +1,98 @@
+use crate::execution::operator::Operator;
+use crate::segment::types::SegmentFlusher;
+use crate::segment::ChromaSegmentFlusher;
+use async_trait::async_trait;
+use chroma_error::ChromaError;
+use chroma_error::ErrorCodes;
+use chroma_types::SegmentFlushInfo;
+use parking_lot::Mutex;
+use thiserror::Error;
+use tracing::trace_span;
+use tracing::Instrument;
+
+#[derive(Error, Debug)]
+pub enum FlushSegmentWriterOperatorError {
+    #[error("Finishing segment writer failed {0}")]
+    FinishSegmentWriterFailed(#[from] Box<dyn ChromaError>),
+    #[error("Segment flusher is missing")]
+    SegmentFlusherMissing,
+}
+
+impl ChromaError for FlushSegmentWriterOperatorError {
+    fn code(&self) -> ErrorCodes {
+        match self {
+            FlushSegmentWriterOperatorError::FinishSegmentWriterFailed(e) => e.code(),
+            FlushSegmentWriterOperatorError::SegmentFlusherMissing => ErrorCodes::Internal,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct FlushSegmentWriterOperator {}
+
+impl FlushSegmentWriterOperator {
+    pub fn new() -> Box<Self> {
+        Box::new(FlushSegmentWriterOperator {})
+    }
+}
+
+#[derive(Debug)]
+pub struct FlushSegmentWriterInput {
+    segment_flusher: Mutex<Option<ChromaSegmentFlusher>>,
```

Comment:
Is a flusher not send/sync?

Reply:
it is, but I need a way to consume the input since `flush(self)` consumes self
and task handlers receive `&input`

Reply:
Ah